src/main/java/address/ui/Ui.java

```diff
@@ -0,0 +1,34 @@
+package address.ui;
+
+import address.MainApp;
+import address.browser.BrowserManager;
+import address.controller.MainController;
+import address.model.ModelManager;
+import address.util.Config;
+import javafx.scene.control.Alert;
+import javafx.stage.Stage;
+
+/**
+ * The UI of the app.
+ */
+public class Ui {
+    MainController mainController;
+
+    public Ui(MainApp mainApp, ModelManager modelManager, Config config){
+        BrowserManager.initializeJxBrowserEnvironment();
```

Comment:
This will fail. It needs to be in the original place because `init` of `MainApp` is special in a sense that it is not in UI thread.

This call needs to be in non-UI thread from the documentation, but it only works if called in `init` of `MainApp` (which is a non-UI thread). I tried to call this in a separate thread at about the same position as here and the application will get stuck because of deadlock (UI is initalizing browser but browser is waiting for UI to be initialized completely).